webxr/idlharness.https.window.js

```diff
@@ -1,17 +1,55 @@
 // META: script=/resources/WebIDLParser.js
 // META: script=/resources/idlharness.js
+// META: timeout=long
 
 'use strict';
 
 // https://immersive-web.github.io/webxr/
 
 idl_test(
   ['webxr'],
-  ['permissions', 'webgl1', 'html', 'dom'],
+  ['permissions', 'webgl1', 'geometry', 'html', 'dom'],
   async idl_array => {
     idl_array.add_objects({
       Navigator: ['navigator'],
       XR: ['navigator.xr'],
+      // TODO: XRSystem
+      XRSession: ['xrSession'],
+      XRRenderState: ['xrRenderState'],
+      // TODO: XRFrame
+      // TODO: XRSpace
+      XRReferenceSpace: ['xrReferenceSpace'],
+      // TODO: XRBoundedReferenceSpace
+      // TODO: XRView
+      // TODO: XRViewport
+      XRRigidTransform: ['new XRRigidTransform()'],
+      // TODO: XRPose
+      // TODO: XRViewerPose
+      // TODO: XRInputSource
+      XRInputSourceArray: ['xrInputSourceArray'],
+      XRWebGLLayer: ['xrWebGLLayer'],
+      WebGLRenderingContextBase: ['webGLRenderingContextBase'],
+      XRSessionEvent: ['xrSessionEvent'],
+      // TODO: XRInputSourceEvent
+      XRInputSourcesChangeEvent: ['xrInputSourcesChangeEvent'],
+      // TODO: XRReferenceSpaceEvent
+      // TODO: XRPermissionStatus
     });
+
+    self.xrSession = await navigator.xr.requestSession('inline');
+    self.xrRenderState = self.xrSession.renderState;
+    self.xrReferenceSpace = await self.xrSession.requestReferenceSpace('viewer');
```

Comment:
Untested, but if the WebXR Test API is set up correctly, something along these lines should work to get an XRFrame:

```js
self.xrFrame = await new Promise((resolve, reject) => {
  self.xrSession.requestAnimationFrame((time, frame) => resolve(frame));
});
```

Note that the XRFrame is only valid for the scope of a rAF callback or while executing an XR event handler callback, so storing it for later use may run into issues.

Reply:
> Untested, but if the WebXR Test API is set up correctly, something along these lines should work to get an XRFrame:

Yes, but my understanding is that right now the WebXR Test API doesn't work in upstream WPT in any browser? (Hence all the red results in https://wpt.fyi/results/webxr?label=experimental&label=master&aligned). That makes developing a change like this trickier, so I'm not inclined to try to do xrFrame yet :)

> Note that the XRFrame is only valid for the scope of a rAF callback or while executing an XR event handler callback, so storing it for later use may run into issues.

It should be ok for idlharness case I hope, since the GC guarantees the object itself will live and all we care about then is walking the prototype I think. And if the spec says that the prototype might change after the rAF is over then I'm really scared!